packages/mui-material/src/List/List.js

```diff
@@ -8,10 +8,16 @@ import ListContext from './ListContext';
 import { getListUtilityClass } from './listClasses';
 
 const useUtilityClasses = (ownerState) => {
-  const { classes, disablePadding, dense, subheader } = ownerState;
+  const { classes, disablePadding, dense, subheader, variant } = ownerState;
 
   const slots = {
-    root: ['root', !disablePadding && 'padding', dense && 'dense', subheader && 'subheader'],
+    root: [
+      'root',
+      variant && variant,
```

Comment:
In case this PR is resumed: the second `variant` here is missing quotes. Also, in other components the variant classes are added like this, so this should probably do the same:

```suggestion
      variant && `variant${capitalize(variant)}`,
```